### PR TITLE
Add thread timestamp to message event

### PR DESCRIPTION
**Pull request by pa3ng**
Status: **POST REVIEW CHANGES DONE**

- Added the `thread_ts` field to the`MessageEvent` struct for being able to reply to threads (docs: https://api.slack.com/docs/message-threading)

- Updated the token verification error message to be more informative.

### CODE REVIEW

**Review by ruudk — APPROVED** [review]

File: slackevents/inner_events.go [hunk 1]
```diff
@@ -12,12 +12,13 @@ type EventsAPIInnerEvent struct {
...
-	TimeStamp      string      `json:"ts"`
-	Channel        string      `json:"channel"`
-	EventTimeStamp json.Number `json:"event_ts"`
+	Type            string      `json:"type"`
+	User            string      `json:"user"`
+	Text            string      `json:"text"`
+	TimeStamp       string      `json:"ts"`
+	Channel         string      `json:"channel"`
+	EventTimeStamp  json.Number `json:"event_ts"`
+	ThreadTimeStamp string      `json:"thread_ts"`
```
Comment (whithajess) [inline slackevents/inner_events.go]: `thread_ts` is not part of the app_mention event: https://api.slack.com/events/app_mention

File: slackevents/parsers.go [hunk 2]
```diff
@@ -158,7 +158,7 @@ func ParseEvent(rawEvent json.RawMessage, opts ...Option) (EventsAPIEvent, error
 	}
 
 	if !cfg.TokenVerified {
-		return EventsAPIEvent{}, errors.New("No")
+		return EventsAPIEvent{}, errors.New("Invalid verification token")
```
Comment (whithajess) [inline slackevents/parsers.go:161]: Haha I guess I am not the only one who thought that should be changed! https://github.com/whithajess/unity/blob/master/hello/main.go#L30

File: slackevents/parsers_test.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ func TestParserOuterCallBackEvent(t *testing.T) {
 				"event": {
 								"type": "app_mention",
 								"event_ts": "1234567890.123456",
+								"thread_ts": "1234567890.123456",
```
Comment (whithajess) [inline slackevents/parsers_test.go]: `thread_ts` is not part of the app_mention event: https://api.slack.com/events/app_mention

File: slackevents/parsers_test.go [hunk 4]
```diff
@@ -82,6 +83,7 @@ func TestThatOuterCallbackEventHasInnerEvent(t *testing.T) {
...
```
Comment (whithajess) [inline slackevents/parsers_test.go]: `thread_ts` is not part of the app_mention event: https://api.slack.com/events/app_mention

**Review by whithajess — CHANGES_REQUESTED** [review]

File: slackevents/parsers.go (hunk 2 above)
Comment (ruudk) [reply slackevents/parsers.go:161]: So that was causing the weird "No" error yesterday 😂

Comment (james-lawrence) [reply slackevents/parsers.go:161]: rofl.

**Comment by james-lawrence** [comment]
will merge one whithajess comments are addressed regarding the tests.

**Comment by pa3ng** [comment]
@james-lawrence addressed whithajess' comments

**Comment by james-lawrence** [comment]
mind rebasing against master? had another PR with event changes. want to make sure it applies cleanly and I'm not familiar enough with the API details to do it at a glance.

**Comment by pa3ng** [comment]
@james-lawrence rebased

**Pull request merged**